src/main/java/teammates/ui/controller/InstructorCommentsPageAction.java

```diff
@@ -73,24 +70,21 @@ public ActionResult execute() throws EntityDoesNotExistException {
             feedbackSessions = getFeedbackSessions();
         }
         
-        data.coursePaginationList = coursePaginationList;
-        data.comments = giverEmailToCommentsMap;
-        data.roster = roster;
-        data.feedbackSessions = feedbackSessions;
-        data.instructorEmail = instructor != null ? instructor.email : "no-email";
-        data.previousPageLink = previousPageLink;
-        data.nextPageLink = nextPageLink;
         int numberOfPendingComments = 0;
         if (!courseId.isEmpty()) {
             numberOfPendingComments = logic.getCommentsForSendingState(courseId, CommentSendingState.PENDING).size() 
                     + logic.getFeedbackResponseCommentsForSendingState(courseId, CommentSendingState.PENDING).size();
         }
-        data.numberOfPendingComments = numberOfPendingComments;
         
         statusToAdmin = "instructorComments Page Load<br>" + 
                 "Viewing <span class=\"bold\">" + account.googleId + "'s</span> comment records " +
                 "for Course <span class=\"bold\">[" + courseId + "]</span>";
-            
+
+        data.init(isViewingDraft, isDisplayArchivedCourse, courseId, courseName, coursePaginationList,
+                  giverEmailToCommentsMap, instructor != null ? instructor.email : "no-email", instructor, 
+                  roster,  feedbackSessions, previousPageLink, nextPageLink, numberOfPendingComments, 
+                  getGiverEmailToGiverNameMap(roster));
```

Comment:
Can this be shortened any further?
From a short analysis, it seems `instructor != null ? instructor.email : "no-email"`, `previousPageLink`, and `nextPageLink` can be derived from the other variables passed in.
Check if the rest are really all used on the page too.